kbase-extension/static/kbase/js/widgets/function_output/kbaseSeqCompView.js

```diff
@@ -0,0 +1,201 @@
+            	    var comparisonsData = [];
+            	    var compTable = $('#'+pref+'comparisons-table');
+		    var row = '<tr><td>Query \\ Reference</td>';
+		    for (var i = 0; i < n_genomes; i++) {
+			var ii = i+1;
+			row += ('<td>'+ii.toString()+'</td>');
+		    }
+		    row += '</tr>';
+		    compTable.append(row);
+		    var minSim = 1;
+		    for (var i = 0; i < data.genome_comparisons.length; i++) {
+			console.log(data.genome_comparisons[i].similarity);
+			if (data.genome_comparisons[i].similarity < minSim) {
+			    minSim = data.genome_comparisons[i].similarity;
+			}
+		    }
+		    // console.log(minSim);
+		    var color = d3.scale.linear()
+		        .domain([minSim, (minSim+1.0)/2, 1.0])
+		        .range(["PaleVioletRed", "pink", "white"]);
+		    var reports = [];
+		    for (var i = 0; i < n_genomes; i++) {
+		    	var ii = i+1;
+			var iname = ii + '. ' + data.genome_names[i];
+			var lp = iname.indexOf('(');
+			if (lp > 0) {
+			    iname = iname.substring(0, lp-1);
+			}
+			var row = '<tr> <td>'+iname+'</td>';
+		    	for (var j = 0; j < n_genomes; j++) {
+		    	    var jj = j+1;
+		    	    comp = data.genome_comparisons.pop();
+			    reports.push(comp.report);
+			    cellId = pref+'_td_'+ii+'_'+jj;
+			    row += '<td><div id="'+cellId+'" style="background-color:'+color(comp.similarity)+'">'+comp.similarity.toString()+'</div></td>';
+			    $('#'+cellId).tooltip({
+				container: "body",
+				title: "tooltip text"
+			    });
+		    	}
+		    	row += '</tr>';
+			compTable.append(row);
+		    }
+
+		    for (var i = 0; i < n_genomes; i++) {
+		    	var ii = i+1;
+		    	for (var j = 0; j < n_genomes; j++) {
+		    	    var jj = j+1;
+		    	    var cellId = pref+'_td_'+ii+'_'+jj;
+		    	    var report = reports.pop();
+		    	    $('#'+cellId).tooltip({
+		    		container: "body",
+				html: true,
+				title: '<div style="font-size: 10px; font-family: Consolas, Menlo, DejaVu Sans Mono, Courier New, monospace, serif; white-space: pre;">'+report+'</div>'
+		    	    });
+		    	}
+		    }
+
+		    console.log(compTable);
+            	}, function(data) {
+            	    container.empty();
+            	    container.append('<p>[Error] ' + data.error.message + '</p>');
+            	});
+            };
+            ready();
+            return this;
+        },
+
+        getData: function() {
+        	return {
+        		type: "NarrativeTempCard",
+        		id: this.ws_name + "." + this.ws_id,
+        		workspace: this.ws_name,
+        		title: "Temp Widget"
+        	};
+        },
+
+        loggedInCallback: function(event, auth) {
+            this.token = auth.token;
+            this.render();
+            return this;
+        },
+
+        loggedOutCallback: function(event, auth) {
+            this.token = null;
+            this.render();
+            return this;
+        },
+
+        uuid: function() {
```

Comment:
There's now a separate module, `util/string` you can use as a dependency that provides uuids. See https://github.com/kbase/narrative/blob/develop/docs/developer/widget_helper_modules.md#utilstring